src/libproxychains.c

```diff
@@ -587,6 +600,51 @@ static int is_v4inv6(const struct in6_addr *a) {
 	return !memcmp(a->s6_addr, "\0\0\0\0\0\0\0\0\0\0\xff\xff", 12);
 }
 
+static int compare_func_int(const void *l, const void *r) {
+    int arg1 = *(const int*)l;
+    int arg2 = *(const int*)r;
+    return (arg1 > arg2) - (arg1 < arg2);
+}
+/* Warning: Linux manual says the third arg is `unsigned int`, but unistd.h says `int`. */
+HOOKFUNC(int, close_range, unsigned first, unsigned last, int flags) {
+    if(!init_l) {
+        /* push back to cache, and delay the execution. */
+        if(close_range_buffer_cnt >= (sizeof close_range_buffer / sizeof *close_range_buffer)) {
+            errno = ENOMEM;
+            return -1;
+        }
+        int i = close_range_buffer_cnt++;
+        close_range_buffer[i].first = first;
+        close_range_buffer[i].last = last;
+        close_range_buffer[i].flags = flags;
+        return errno = 0;
+    }
+	if(proxychains_resolver != DNSLF_RDNS_THREAD) return true_close_range(first, last, flags);
+
+	/* prevent rude programs (like ssh) from closing our pipes */
+    int res = 0, uerrno = 0;
+    int protected_fds[] = {req_pipefd[0], req_pipefd[1], resp_pipefd[0], resp_pipefd[1]};
+    int protected_fds_size = sizeof protected_fds / sizeof *protected_fds;
```

Comment:
i'm not sure `sizeof *protected` is right here as it's an array, not a pointer.
should be `sizeof xxx[0]`

Reply:
Copied from this example: https://en.cppreference.com/w/c/algorithm/qsort
I believe it's ok.